src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/reporting/ndmis/performance/NdmisPerformanceReportJobConfiguration.kt

```diff
@@ -57,45 +106,59 @@ class NdmisPerformanceReportJobConfiguration(
       .start(ndmisWriteReferralToCsvStep)
       .next(ndmisWriteComplexityToCsvStep)
       .next(ndmisWriteAppointmentToCsvStep)
+      .next(pushToS3Step)
       .build()
   }
 
   @Bean
   fun ndmisWriteReferralToCsvStep(
     ndmisReader: HibernateCursorItemReader<Referral>,
     processor: ReferralsProcessor,
+    writer: FlatFileItemWriter<ReferralsData>,
   ): Step {
     return stepBuilderFactory.get("ndmisWriteReferralToCsvStep")
       .chunk<Referral, ReferralsData>(chunkSize)
       .reader(ndmisReader)
       .processor(processor)
-      .writer(LoggingWriter<ReferralsData>())
+      .writer(writer)
       .build()
   }
 
   @Bean
   fun ndmisWriteComplexityToCsvStep(
     ndmisReader: HibernateCursorItemReader<Referral>,
     processor: ComplexityProcessor,
+    writer: FlatFileItemWriter<Collection<ComplexityData>>,
   ): Step {
     return stepBuilderFactory.get("ndmisWriteComplexityToCsvStep")
       .chunk<Referral, List<ComplexityData>>(chunkSize)
       .reader(ndmisReader)
       .processor(processor)
-      .writer(LoggingWriter<List<ComplexityData>>())
+      .writer(writer)
       .build()
   }
 
   @Bean
   fun ndmisWriteAppointmentToCsvStep(
     ndmisReader: HibernateCursorItemReader<Referral>,
     processor: AppointmentProcessor,
+    writer: FlatFileItemWriter<Collection<AppointmentData>>,
   ): Step {
     return stepBuilderFactory.get("ndmisWriteAppointmentToCsvStep")
       .chunk<Referral, List<AppointmentData>>(chunkSize)
       .reader(ndmisReader)
       .processor(processor)
-      .writer(LoggingWriter<List<AppointmentData>>())
+      .writer(writer)
       .build()
   }
+
+  private val pushToS3Step = stepBuilderFactory["pushToS3Step"]
+    .tasklet { _: StepContribution, _: ChunkContext ->
+      listOf(referralReportPath, complexityReportPath, appointmentsReportPath).forEach { path ->
+        s3Service.publishFileToS3(ndmisS3Bucket, path, "export/csv/reports/")
```

Comment:
the current path we are putting the csv file is `dfinterventions/dfi/csv/reports/`. do we need to put these in the same place? leigh mentioned we will have some overlap, so probably not. but we should ask where the ndmis team want them in the bucket.

Reply:
the ticket says: 
Upload those files to S3 bucket (confirmed)
export/csv/reports/crs_performance_report-v2-referrals.csv